### PR TITLE
ensure multi source spec is handled

### PR DESCRIPTION
**Pull request by nabuskey**
Fixes an issue when you are using multi sources in ArgoCD, the repo url was not updated correctly. It was because we were not updating the correct object.

### CODE REVIEW

File: pkg/controllers/custompackage/test/resources/customPackages/testDir/app2.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  name: my-app2
+  namespace: argocd
+spec:
+  destination:
+    namespace: my-app2
+    server: "https://kubernetes.default.svc"
+  sources:
+    - repoURL: cnoe://busybox
+      targetRevision: HEAD
+      path: "."
```
Comment (cmoulliard) [inline pkg/controllers/custompackage/test/resources/customPackages/testDir/app2.yaml:13]: Sorry if I missed something but this test case only includes one source and not (as title mentions) several sources !

Comment (nabuskey) [reply pkg/controllers/custompackage/test/resources/customPackages/testDir/app2.yaml:13]: Yeah it's meant to represent when the sources field is used. Which is not exactly the same as what is described in the title. I'll update the test.

**Comment by nabuskey** [comment]
Updated it to further handle a case where the same directory is specified twice in the spec. e.g.

```yaml
spec:
  sources:
    - repoURL: cnoe://app2
      targetRevision: HEAD
      path: "one"
    - repoURL: cnoe://app2
      targetRevision: HEAD
      path: "two"
```

**Review by cmoulliard — APPROVED** [review]

File: pkg/controllers/custompackage/controller.go [hunk 2]
```diff
@@ -190,8 +190,13 @@ func (r *Reconciler) reconcileGitRepo(ctx context.Context, resource *v1alpha1.Cu
 		}
 		return nil
 	})
+	// it's possible for an application to specify the same directory multiple times in the spec.
+	// if there is a repository already created for this package, no further action is necessary.
+	if !errors.IsAlreadyExists(err) {
```
Comment (jessesanford) [inline pkg/controllers/custompackage/controller.go:195]: Good catch

**Review by jessesanford — APPROVED** [review]
LGTM

**Pull request merged**